test/extensions/filters/http/ext_proc/ext_proc_integration_common.cc

```diff
@@ -838,40 +850,80 @@ void ExtProcIntegrationTest::serverSendTrailerRespDuplexStreamed() {
-            "@type": type.googleapis.com/envoy.extensions.filters.http.composite.v3.Composite
-        xds_matcher:
-          matcher_tree:
-            input:
-              name: request-headers
-              typed_config:
-                "@type": type.googleapis.com/envoy.type.matcher.v3.HttpRequestHeaderMatchInput
-                header_name: match-header
-            exact_match_map:
-              map:
-                match:
-                  action:
-                    name: composite-action
-                    typed_config:
-                      "@type": type.googleapis.com/envoy.extensions.filters.http.composite.v3.ExecuteFilterAction
-                      typed_config:
-                        name: envoy.filters.http.ext_proc
-                        typed_config:
-                          "@type": type.googleapis.com/envoy.extensions.filters.http.ext_proc.v3.ExternalProcessor
-                          grpc_service:
-                            envoy_grpc:
-                              cluster_name: ext_proc_server_0
-                            timeout: 300s
-
-    )EOF"),
-                               true);
+void ExtProcIntegrationTest::
+    prependExtProcCompositeFilterMatchOnRequestHeaders() {
+  envoy::extensions::filters::network::http_connection_manager::v3::HttpFilter
+      composite_filter;
+  composite_filter.set_name("composite");
+
+  envoy::extensions::common::matching::v3::ExtensionWithMatcher
+      extension_with_matcher;
+  auto* extension_config = extension_with_matcher.mutable_extension_config();
+  extension_config->set_name("composite");
+  envoy::extensions::filters::http::composite::v3::Composite composite_config;
+  extension_config->mutable_typed_config()->PackFrom(composite_config);
+
+  auto* matcher_tree =
+      extension_with_matcher.mutable_xds_matcher()->mutable_matcher_tree();
+  auto* input = matcher_tree->mutable_input();
+  input->set_name("request-headers");
+  envoy::type::matcher::v3::HttpRequestHeaderMatchInput header_match_input;
+  header_match_input.set_header_name("match-header");
+  input->mutable_typed_config()->PackFrom(header_match_input);
+
+  envoy::extensions::filters::http::composite::v3::ExecuteFilterAction
+      execute_filter_action;
+  auto* typed_config = execute_filter_action.mutable_typed_config();
+  typed_config->set_name("envoy.filters.http.ext_proc");
+  typed_config->mutable_typed_config()->PackFrom(proto_config_);
+
+  auto& on_match =
+      (*matcher_tree->mutable_exact_match_map()->mutable_map())["match"];
+  on_match.mutable_action()->set_name("composite-action");
+  on_match.mutable_action()->mutable_typed_config()->PackFrom(
+      execute_filter_action);
+
+  composite_filter.mutable_typed_config()->PackFrom(extension_with_matcher);
+  config_helper_.prependFilter(
+      MessageUtil::getJsonStringFromMessageOrError(composite_filter), true);
+}
+
+void ExtProcIntegrationTest::
+    prependExtProcCompositeFilterMatchOnResponseHeaders() {
```

Comment:
It looks to me you can share common code for these two test functions. Otherwise LGTM